network/p2p.py

```
@@ -0,0 +1,34 @@
+
+    async def broadcast_block(self, block):
+        msg = json.dumps({"type": "block", "data": block.to_dict()})
+        print(f"Network: Broadcasting Block #{block.index}")
+        # await self.pubsub.publish("minichain-global", msg.encode())
+
+    async def handle_message(self, msg):
+        """Callback when p2p message is received"""
+        try:
+            data = json.loads(msg.data.decode())
+            await self.handler_callback(data)
+        except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)